### PR TITLE
LibWeb: Bring performance.now() closer to spec

### PR DESCRIPTION
**Pull request by stelar7**
Fixes a few WPT tests on https://wpt.live/dom/events/Event-timestamp-high-resolution.html

### CODE REVIEW

**Comment by awesomekling** [comment]
Can we add in-tree tests?

File: Userland/Libraries/LibWeb/HighResolutionTime/Performance.cpp [hunk 1]
```diff
@@ -65,11 +65,20 @@ JS::GCPtr<NavigationTiming::PerformanceNavigation> Performance::navigation()
...
 {
+    // FIXME: The timeOrigin attribute MUST return the number of milliseconds in the duration returned by get time origin timestamp for the relevant global object of this.
     return static_cast<double>(m_timer.origin_time().nanoseconds()) / 1e6;
 }
 
+// https://w3c.github.io/hr-time/#now-method
+double Performance::now() const
+{
+    // The now() method MUST return the number of milliseconds in the current high resolution time given this's relevant global object (a duration).
+    return current_high_resolution_time(this->realm().global_object());
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HighResolutionTime/Performance.cpp:79]: You should be able to use `HTML::relevant_global_object(*this)` here.

Comment (stelar7) [reply Userland/Libraries/LibWeb/HighResolutionTime/Performance.cpp]: Fixed 👍

File: Tests/LibWeb/Text/input/performance-now.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html>
+<script src="include.js"></script>
+<script>
+    test(() => {
+        let before = performance.now();
+        let event = new window["MouseEvent"]('test');
```
Comment (tcl3) [inline Tests/LibWeb/Text/input/performance-now.html:6]: It should be fine to use `new MouseEvent('test')` here.

Comment (stelar7) [reply Tests/LibWeb/Text/input/performance-now.html]: Fixed 👍

File: Userland/Libraries/LibWeb/HighResolutionTime/TimeOrigin.cpp [hunk 3]
```diff
@@ -14,9 +14,32 @@ namespace Web::HighResolutionTime {
...
+
+    // To get time origin timestamp, given a global object global, run the following steps, which return a duration:
+    // FIXME: 1. Let timeOrigin be global's relevant settings object's time origin.
+    auto time_origin = 0;
+
+    // Each group of environment settings objects that could possibly communicate in any way
+    // has an estimated monotonic time of the Unix epoch, a moment on the monotonic clock,
+    // whose value is initialized by the following steps:
+
+    // !. Let wall time be the wall clock's unsafe current time.
```
Comment (nico) [inline Userland/Libraries/LibWeb/HighResolutionTime/TimeOrigin.cpp:27]: !

**Pull request merged**